terraform/src/index.js

```diff
@@ -0,0 +1,162 @@
+                        LogGroupName: eventPayload.logGroup,
+                        LogStreamName: eventPayload.logStream,
+                        TaskGuid: taskGuid,
+                        Layer0: {
+                            ContainerName: containerName,
+                            ServiceId: serviceId,
+                            EnvironmentId: environmentId,
+                            Instance: instanceName
+                        }
+                    },
+                    source: environmentId + ':' + serviceId,
+                    sourcetype: "layer0_service",
+                    host: 'layer0',
+                    time: e.timestamp
+                };
+
+                return writeToStream(JSON.stringify(data));
+            });
+
+            return Promise.all(writeToStreams);
+        })
+        .then(() => {
+            callback(null, 'forwarding logs to stream complete');
+        })
+        .catch(err => {
+            console.log(err);
+            callback(err);
+        });
+  };
+
+function writeToStream(data) {
+    const params = {
+        Data: data,
+        PartitionKey: getRandomInt(32).toString(),
+        StreamName: process.env.KINESIS_STREAM_NAME
+    };
+
+    return kinesis.putRecord(params).promise()
+        .then(() => { })
+        .catch(err => {
+            console.log(err);
+        });
+}
+
+function parseEventData(event) {
+    return new Promise((resolve, reject) => {
+        const payload = new Buffer(event.awslogs.data, 'base64');
+        zlib.gunzip(payload, function (error, result) {
+            if (error) {
+                reject(error);
+            } else {
+                resolve(JSON.parse(result.toString('utf8')));
+            }
+        });
+    });
+}
+
+function getRandomInt(max) {
+    return Math.floor(Math.random() * Math.floor(max));
+}
+
+// extract instance name from the log group name
+function parseInstanceName(logGroupName) {
+    // log group name = l0-<instance_name>
+    let instanceName = 'unknown';
+    const logNameParts = logGroupName.split('-');
+
+    if (logNameParts.length == 2) {
+        instanceName = logNameParts[1];
+    }
+
+    return instanceName;
+}
+
+// extract container name from stream name
+function parseContainerName(logStreamName) {
+    //streamName = l0-<instance_name>/l0/<env_name>/<task_guid>
+    let containerName = 'unknown';
+    const streamNameParts = logStreamName.split('/');
+
+    if (streamNameParts.length == 3) {
+        containerName = streamNameParts[1];
+    }
+
+    return containerName;
+}
+
+// extract task guid stream name
+function parseTaskGuid(logStreamName) {
+    //streamName = l0/<env_name>/<task_guid>
+    let taskGuid = 'unknown';
+    const streamNameParts = logStreamName.split('/');
+
+    if (streamNameParts.length == 3) {
+        taskGuid = streamNameParts[2];
+    }
+
+    return taskGuid;
+}
+
+// attempt to determine the severity based on the content
+function extractSeverity(logText) {
```

Comment:
Are these severity levels meant to target log output in v0.10.xv specifically?

Reply:
These are just arbitrary levels. Our current logging library (ImsHealth.Logging) has these levels so all the events that are logged from there have one of these severity levels set. This is more just being consistent with the other structured log data we are currently sending to splunk.